clang/lib/CIR/CodeGen/CIRGenStmt.cpp

```diff
@@ -383,6 +383,16 @@ mlir::LogicalResult CIRGenFunction::buildIfStmt(const IfStmt &S) {
       if (buildStmt(S.getInit(), /*useCurrentScope=*/true).failed())
         return mlir::failure();
 
+    // Handle "if constexpr" explicitly here to avoid generating some
+    // ill-formed code since in CIR the "if" are no longer simplified
+    // in this lambda like in Clang but postponed to other MLIR
+    // passes.
+    if (S.isConstexpr())
+      if (auto ndc = S.getNondiscardedCase(getContext())) {
```

Comment:
What should happen if the stmt is a constexpr stmt and `getNondiscardedCase` returns an empty/invalid result?

Reply:
Bad things happen. :-( Good catch!

Reply:
I misunderstood this API. Even if a `std::optional` has a value, it can be a `nullptr` instead of no value. Confusing.